skbeam/core/recip.py

```diff
@@ -342,3 +342,37 @@ def gisaxs(incident_beam, reflected_beam, pixel_size, detector_size,
     qr = np.sqrt(qx**2 + qy**2)
 
     return gisaxs_output(alpha_i, theta_f, alpha_f, tilt_angle, qx, qy, qz, qr)
+
+
+def generate_q_bins(rmax, qmax, pixel_size, distance, wavelength):
+    """
+    Generate the Q bins at the resolution of the detector
```

Comment:
Almost, the function does generate the non uniform-width bins. However, the mapping is to single rings. Dealing with single pixels is hard, as the square nature of the pixels conflicts with the annular nature of the diffraction. For example, a pixel on a 0 degree angle from the point of incidence would have a better resolution than on at 45 degrees, as the 45 degree pixel is on its diagonal which is longer. Technically, this correctly maps the resolution only for pixels which are straight up, down, left or right from the point of incidence. However, the deviation caused by the pixels off axis are small and thus neglected. To properly capture this off axis behavior one might have to use a split pixel method, so that each subsection of a pixel can be put with subpixels at the same Q value. However, Simon's group have shown that this introduces co-variances into the scattering data.